### ISSUE TITLE
None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by WHQ1111** [Lyken17/pytorch-memonger#4]
Thank you for sharing this work. But when I use it at ResNet, it can't work. The reason maybe the middle layer of ResNet have requires_grad=False.
```
        out = self.layer1(out)
        out = self.layer2(out)
        out = self.layer3(out)
        out = self.layer4(out)
```
When out is feed to layer2, it have requires_grad=False so layer1's parameters can't be optimized. The same problem happen to layer2, layer3 and layer4. And I get
`         UserWarning: None of the inputs have requires_grad=True. Gradients will be None`
How to solve this problem?
